rustls/src/server/verifier_builder.rs

```diff
@@ -0,0 +1,274 @@
+use std::sync::Arc;
+
+use crate::server::UnparsedCertRevocationList;
+use crate::verify::{AnonymousClientPolicy, ClientCertVerifier, WebPkiClientVerifier};
+use crate::{CertRevocationListError, RootCertStore};
+
+/// A builder for configuring a `webpki` client certificate verifier.
+///
+/// For more information, see the [`WebPkiClientVerifier`] documentation.
+#[derive(Debug, Clone)]
+pub struct ClientCertVerifierBuilder {
+    roots: RootCertStore,
+    crls: Vec<webpki::OwnedCertRevocationList>,
+    anon_policy: AnonymousClientPolicy,
+}
+
+impl ClientCertVerifierBuilder {
+    pub(crate) fn new(roots: RootCertStore) -> Result<Self, RootsRequiredError> {
+        match roots.is_empty() {
+            true => Err(RootsRequiredError(())),
+            false => Ok(Self {
+                roots,
+                crls: Vec::new(),
+                anon_policy: AnonymousClientPolicy::Forbid,
+            }),
+        }
+    }
+
+    /// Add additional `roots` to use to verify client certificates.
+    ///
+    /// All clients must provide a client certificate unless you have allowed unauthenticated
+    /// clients by calling [allow_unauthenticated()][ClientCertVerifierBuilder::allow_unauthenticated]
+    /// on the builder.
+    pub fn with_roots(mut self, roots: RootCertStore) -> Self {
+        self.roots
+            .add_server_trust_anchors(roots.roots.into_iter());
+        Self {
+            roots: self.roots,
+            crls: self.crls,
+            anon_policy: self.anon_policy,
+        }
+    }
+
+    /// Verify the revocation state of presented client certificates against the provided
+    /// certificate revocation lists (CRLs). Calling `with_crls` multiple time will merge the
```

Comment:
"merge the CRL[ist]s" is actually not what happens, right? I think a better way to describe this would be more like "appends the given further CRLs to the existing collection".

Reply:
Good catch. Updated the description to say they're appended vs merged.